File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Components/CuratedCollectionsView.swift
import SwiftUI

struct CuratedCollectionsView: View {
    let collections: [CollectionItem]
    
    // Model specific to this view
    struct CollectionItem: Decodable, Identifiable {
        let id = UUID()
        let title: String
        let subtitle: String
        let backgroundColor: String
        let headerImage: String?
        let items: [SubItem]
        
        enum CodingKeys: String, CodingKey {
            case title, subtitle, backgroundColor, headerImage, items
        }
        
        struct SubItem: Decodable, Identifiable {
            let id = UUID()
            let name: String
            let image: String
            let bgColor: String
            let actionUrl: String
            
            enum CodingKeys: String, CodingKey {
                case name, image, bgColor, actionUrl
            }
        }
    }
    
    var body: some View {
        VStack(spacing: 24) {
            ForEach(collections) { collection in
                VStack(alignment: .leading, spacing: 16) {
                    // Start of Card Content
                    
                    // Header Section with Image
                    HStack(alignment: .top) {
                        VStack(alignment: .leading, spacing: 4) {
                            Text(collection.title)
                                .font(.system(size: 22, weight: .bold))
                                .foregroundColor(Color(hex: "#1F2937")) // Dark Gray
                            
                            Text(collection.subtitle)
                                .font(.system(size: 14))
                                .foregroundColor(Color(hex: "#4B5563")) // Medium Gray
                        }
                        
                        Spacer()
                        
                        // Header Image (Floating overlap effect)
                        if let imgStr = collection.headerImage, let url = URL(string: imgStr) {
                            AsyncImage(url: url) { image in
                                image
                                    .resizable()
                                    .aspectRatio(contentMode: .fit)
                            } placeholder: {
                                Color.white.opacity(0.5)
                            }
                            .frame(width: 70, height: 70)
                            // Add a subtle rotation or shadow for "pop"
                            .rotationEffect(.degrees(5))
                            .shadow(color: .black.opacity(0.1), radius: 5, x: 2, y: 2)
                        }
                    }
                    .padding(.horizontal, 20)
                    .padding(.top, 20)
                    
                    // Items Grid (2x2)
                    LazyVGrid(columns: [
                        GridItem(.flexible(), spacing: 16),
                        GridItem(.flexible(), spacing: 16)
                    ], spacing: 16) {
                        ForEach(collection.items) { item in
                            Button(action: {
                                print("Navigate to: \(item.actionUrl)")
                            }) {
                                HStack(spacing: 12) {
                                    // Icon Container
                                    ZStack {
                                        Circle()
                                            .fill(Color(hex: item.bgColor).opacity(0.8))
                                            .frame(width: 48, height: 48)
                                        
                                        AsyncImage(url: URL(string: item.image)) { image in
                                            image
                                                .resizable()
                                                .aspectRatio(contentMode: .fit)
                                        } placeholder: {
                                            ProgressView()
                                                .scaleEffect(0.5)
                                        }
                                        .frame(width: 28, height: 28)
                                    }
                                    
                                    // Item Name
                                    Text(item.name)
                                        .font(.system(size: 14, weight: .semibold))
                                        .foregroundColor(Color(hex: "#374151"))
                                        .lineLimit(1)
                                        .minimumScaleFactor(0.9)
                                    
                                    Spacer()
                                }
                                .padding(8)
                                .background(Color.white.opacity(0.6))
                                .cornerRadius(12)
                            }
                        }
                    }
                    .padding(.horizontal, 20)
                    .padding(.bottom, 24)
                }
                .background(Color(hex: collection.backgroundColor))
                .cornerRadius(24)
                .shadow(color: Color.black.opacity(0.05), radius: 10, x: 0, y: 4)
                .padding(.horizontal, 16)
            }
        }
        .padding(.vertical, 8)
    }
}

File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/ui/components/CuratedCollectionsView.kt
package com.ecommerceearn.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage

@Composable
fun CuratedCollectionsView(collections: List<Map<String, Any>>) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 8.dp),
        verticalArrangement = Arrangement.spacedBy(24.dp)
    ) {
        collections.forEach { collection ->
            val title = collection["title"] as? String ?: ""
            val subtitle = collection["subtitle"] as? String ?: ""
            val backgroundColorHex = collection["backgroundColor"] as? String ?: "#FFFFFF"
            val headerImage = collection["headerImage"] as? String
            val itemsRaw = collection["items"] as? List<*>
            val items = itemsRaw?.mapNotNull { it as? Map<String, String> } ?: emptyList()

            // Card Container
            Box(
                modifier = Modifier
                    .padding(horizontal = 16.dp)
                    .fillMaxWidth()
                    .background(
                        color = safeParseColor(backgroundColorHex),
                        shape = RoundedCornerShape(24.dp)
                    )
            ) {
                Column(
                    modifier = Modifier.padding(bottom = 24.dp)
                ) {
                    // Header Section
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(start = 20.dp, end = 20.dp, top = 20.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.Top
                    ) {
                        Column(modifier = Modifier.weight(1f)) {
                            Text(
                                text = title,
                                fontSize = 22.sp,
                                fontWeight = FontWeight.Bold,
                                color = Color(0xFF1F2937)
                            )
                            Spacer(modifier = Modifier.height(4.dp))
                            Text(
                                text = subtitle,
                                fontSize = 14.sp,
                                color = Color(0xFF4B5563)
                            )
                        }

                        if (headerImage != null) {
                            AsyncImage(
                                model = headerImage,
                                contentDescription = null,
                                modifier = Modifier
                                    .size(70.dp)
                                    .rotate(5f)
                                    .shadow(elevation = 5.dp, shape = RoundedCornerShape(8.dp)),
                                contentScale = ContentScale.Fit
                            )
                        }
                    }

                    Spacer(modifier = Modifier.height(16.dp))

                    // 2x2 Grid Layout
                    Column(
                        modifier = Modifier.padding(horizontal = 20.dp),
                        verticalArrangement = Arrangement.spacedBy(16.dp)
                    ) {
                        val rows = items.chunked(2)
                        rows.forEach { rowItems ->
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.spacedBy(16.dp)
                            ) {
                                rowItems.forEach { item ->
                                    CuratedItemCard(item, modifier = Modifier.weight(1f))
                                }
                                // Fill empty space if odd number
                                if (rowItems.size < 2) {
                                    Spacer(modifier = Modifier.weight(1f))
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

@Composable
fun CuratedItemCard(item: Map<String, String>, modifier: Modifier = Modifier) {
    val name = item["name"] ?: ""
    val image = item["image"] ?: ""
    val bgColor = item["bgColor"] ?: "#EEEEEE"

    Row(
        modifier = modifier
            .background(Color.White.copy(alpha = 0.6f), RoundedCornerShape(12.dp))
            .padding(8.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        // Icon Container
        Box(
            contentAlignment = Alignment.Center
        ) {
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .clip(CircleShape)
                    .background(safeParseColor(bgColor).copy(alpha = 0.8f))
            )
            AsyncImage(
                model = image,
                contentDescription = null,
                modifier = Modifier.size(28.dp),
                contentScale = ContentScale.Fit
            )
        }

        Text(
            text = name,
            fontSize = 14.sp,
            fontWeight = FontWeight.SemiBold,
            color = Color(0xFF374151),
            maxLines = 1,
            modifier = Modifier.weight(1f)
        )
    }
}


